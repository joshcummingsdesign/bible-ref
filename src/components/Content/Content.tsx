import {parseBookId} from '@/helpers/parseBookId';
import {Passage} from '@/types';
import styles from './styles.module.scss';

interface Props {
    passages: Passage[] | null;
}

export const Content = ({passages}: Props) => (
    <section className={styles.content}>
        {passages === null ? (
            <p>No passages found.</p>
        ) : (
            <>
                {passages.map((passage, i) => (
                    <div key={passage.id}>
                        <h2>{parseBookId(passage.bookId)!.title}</h2>
                        <div key={passage.id} dangerouslySetInnerHTML={{__html: passage.content}} />
                    </div>
                ))}
            </>
        )}
    </section>
);
