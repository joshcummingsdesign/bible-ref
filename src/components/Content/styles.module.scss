@use '../../styles/fonts' as *;
@use '../../styles/grid' as *;

.content {
    padding: $s-md 0;

    p {
        margin: $s-md 0;
    }

    // Section headings
    [class~="s1"],
    [class~="s2"] {
        font-weight: bold;
    }

    // Section titles
    [class~="d"] {
        text-transform: uppercase;
        letter-spacing: 1px;
        font-size: 0.8125rem;
        font-style: italic;
        margin-top: -20px;
        font-weight: $fw-bold;
    }

    // Verse numbers
    [class~="v"] {
        vertical-align: super;
        font-size: 0.75rem;
        font-weight: $fw-bold;
    }

    // References
    [class~="r"] {
        display: none;
    }

    // Poetry
    [class~="q0"], // odd
    [class~="q1"] { // even
        margin-top: -20px;
        margin-left: 20px;
    }

    [class~="q1"] { // even
        margin-left: 40px;
    }

    // Selah
    [class~="qr"] {
        text-align: right;
        float: right;
        margin-top: -57.25px;
        font-style: italic;
    }

    // First line of poetry
    h3,
    p:not([class^="q"]) {
        & + [class^="q"] {
            margin-top: $s-md;
        }
    }

    @for $i from 1 through 6 {
        h#{$i} {
            margin: $s-md 0;
        }
    }
}
