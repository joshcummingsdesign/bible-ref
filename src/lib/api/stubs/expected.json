[
    {
        "id": "PSA.1.1-PSA.4.3",
        "orgId": "PSA.1.1-PSA.4.3",
        "bibleId": "de4e12af7f28f599-01",
        "bibleName": "BSB",
        "bookId": "PSA",
        "chapterIds": [
            "PSA.1",
            "PSA.2",
            "PSA.3",
            "PSA.4"
        ],
        "reference": "Psalms 1:1-4:3",
        "content": "<h3>Psalm 1</h3><p class=\"s1\">Heading 1</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 1:1\">1</span>Verse 1:1</p><h3>Psalm 2</h3><p class=\"s2\">Heading 2</p><p class=\"d\">Description 1</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 2:18\">18</span>Verse 2:18?</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 2:19\">19</span>Verse 2:19.</p><h3>Psalm 3</h3><p class=\"p\"><span class=\"v\" data-sid=\"PSA 3:24\">24</span>Verse 3:24</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 3:25\">25</span>Verse 3:25</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 3:26\">26</span>Verse 3:26</p><h3>Psalm 4</h3><p class=\"d\">Description 2</p><p class=\"p\"><span class=\"v\" data-sid=\"PSA 4:3\">3</span>Verse 4:3</p><p class=\"q0\"><span class=\"v\" data-sid=\"PSA 4:4\">4</span>Verse 4:4</p><p class=\"q1\"><span class=\"v\" data-sid=\"PSA 4:5\">5</span>Verse 4:5</p>",
        "verseCount": 7,
        "copyright": "PUBLIC DOMAIN except in the United Kingdom, where a Crown Copyright applies to printing the KJV. See http://www.cambridge.org/about-us/who-we-are/queens-printers-patent"
    }
]
