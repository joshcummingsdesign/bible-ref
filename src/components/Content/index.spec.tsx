import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';
import {Content} from './Content';
import expected from '@/lib/api/stubs/expected.json';

it('should handle none found', () => {
    render(<Content passages={null} />);

    const nonFound = screen.getByText('No passages found.');

    expect(nonFound).toBeInTheDocument();
});

it('should handle empty passages', () => {
    render(<Content passages={[]} />);

    const nonFound = screen.getByText('No passages found.');

    expect(nonFound).toBeInTheDocument();
});

it('should show passages', () => {
    render(<Content passages={expected} />);

    const title = screen.getByRole('heading', {level: 2});
    const headings = screen.getAllByRole('heading', {level: 3});

    expect(title).toHaveTextContent('John');
    expect(headings[0]).toHaveTextContent('John 1');
    expect(headings[1]).toHaveTextContent('John 2');
});
