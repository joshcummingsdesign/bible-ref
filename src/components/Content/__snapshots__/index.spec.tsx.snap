// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show passages 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="content"
      >
        <div>
          <h2>
            Psalms
          </h2>
          <div>
            <h3>
              Psalm 1
            </h3>
            <p
              class="s1"
            >
              Heading 1
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 1:1"
              >
                1
              </span>
              Verse 1:1
            </p>
            <h3>
              Psalm 2
            </h3>
            <p
              class="s2"
            >
              Heading 2
            </p>
            <p
              class="d"
            >
              Description 1
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 2:18"
              >
                18
              </span>
              Verse 2:18?
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 2:19"
              >
                19
              </span>
              Verse 2:19.
            </p>
            <h3>
              Psalm 3
            </h3>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 3:24"
              >
                24
              </span>
              Verse 3:24
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 3:25"
              >
                25
              </span>
              Verse 3:25
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 3:26"
              >
                26
              </span>
              Verse 3:26
            </p>
            <h3>
              Psalm 4
            </h3>
            <p
              class="d"
            >
              Description 2
            </p>
            <p
              class="p"
            >
              <span
                class="v"
                data-sid="PSA 4:3"
              >
                3
              </span>
              Verse 4:3
            </p>
            <p
              class="q0"
            >
              <span
                class="v"
                data-sid="PSA 4:4"
              >
                4
              </span>
              Verse 4:4
            </p>
            <p
              class="q1"
            >
              <span
                class="v"
                data-sid="PSA 4:5"
              >
                5
              </span>
              Verse 4:5
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="content"
    >
      <div>
        <h2>
          Psalms
        </h2>
        <div>
          <h3>
            Psalm 1
          </h3>
          <p
            class="s1"
          >
            Heading 1
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 1:1"
            >
              1
            </span>
            Verse 1:1
          </p>
          <h3>
            Psalm 2
          </h3>
          <p
            class="s2"
          >
            Heading 2
          </p>
          <p
            class="d"
          >
            Description 1
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 2:18"
            >
              18
            </span>
            Verse 2:18?
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 2:19"
            >
              19
            </span>
            Verse 2:19.
          </p>
          <h3>
            Psalm 3
          </h3>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 3:24"
            >
              24
            </span>
            Verse 3:24
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 3:25"
            >
              25
            </span>
            Verse 3:25
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 3:26"
            >
              26
            </span>
            Verse 3:26
          </p>
          <h3>
            Psalm 4
          </h3>
          <p
            class="d"
          >
            Description 2
          </p>
          <p
            class="p"
          >
            <span
              class="v"
              data-sid="PSA 4:3"
            >
              3
            </span>
            Verse 4:3
          </p>
          <p
            class="q0"
          >
            <span
              class="v"
              data-sid="PSA 4:4"
            >
              4
            </span>
            Verse 4:4
          </p>
          <p
            class="q1"
          >
            <span
              class="v"
              data-sid="PSA 4:5"
            >
              5
            </span>
            Verse 4:5
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
